typescript/infra/scripts/deploy.ts

```diff
@@ -136,6 +156,157 @@ async function main() {
   await deployWithArtifacts(config, deployer, cache, fork, agentConfig);
 }
 
+const TEST_QUERY_SENDER_FACTORIES = {
+  TestQuerySender: new TestQuerySender__factory(),
+};
+
+type TestQuerySenderConfig = { queryRouterAddress: string };
+
+class TestQuerySenderDeployer extends HyperlaneDeployer<
+  TestQuerySenderConfig,
+  typeof TEST_QUERY_SENDER_FACTORIES
+> {
+  constructor(multiProvider: MultiProvider, protected igp: HyperlaneIgp) {
+    super(multiProvider, TEST_QUERY_SENDER_FACTORIES);
+  }
+
+  async deployContracts(chain: ChainName, config: TestQuerySenderConfig) {
+    const TestQuerySender = await this.deployContract(
+      chain,
+      'TestQuerySender',
+      [],
+      [
+        config.queryRouterAddress,
+        this.igp.getContracts(chain).interchainGasPaymaster.address,
+      ],
+    );
+    return {
+      TestQuerySender,
+    };
+  }
+}
+
+const TEST_RECIPIENT_DEPLOYER_FACTORIES = {
+  TestRecipient: new TestRecipient__factory(),
+  TestTokenRecipient: new TestTokenRecipient__factory(),
+};
+
+class TestRecipientDeployer extends HyperlaneDeployer<
+  never,
+  typeof TEST_RECIPIENT_DEPLOYER_FACTORIES
+> {
+  constructor(multiProvider: MultiProvider) {
+    super(multiProvider, TEST_RECIPIENT_DEPLOYER_FACTORIES);
+  }
+
+  async deployContracts(chain: ChainName) {
+    const TestRecipient = await this.deployContract(chain, 'TestRecipient', []);
+    const TestTokenRecipient = await this.deployContract(
+      chain,
+      'TestTokenRecipient',
+      [],
+    );
+    return {
+      TestRecipient,
+      TestTokenRecipient,
+    };
+  }
+}
+
+async function deployWithArtifacts<Config>(
```

Comment:
I personally like the pattern on having a script generally be just a `main` fn and then having helper functions live outside the script

also allows for functions like this or `writeAgentConfig` down below to be easily imported / used by other scripts at some point if needed